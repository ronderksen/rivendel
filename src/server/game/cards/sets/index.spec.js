import uuid from 'uuid';
import PlayerCards from './index';

jest.mock('uuid');
uuid.v1 = jest.fn().mockReturnValue('xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx')

describe('PlayerCards', () => {
  it('should load the player cards', () => {
    expect(PlayerCards['116-001'].name).toBe('Aragorn');
  });

  it('should be possible to instantiate card with given card data', () => {
    const cardData = {
      "name": "Aragorn",
      "label": "Aragorn",
      "id": "33",
      "unique": "Yes",
      "type": "Hero",
      "deck": "Hero",
      "sphere": "Leadership",
      "encounter": "None",
      "cost": 0,
      "textcost": "",
      "th": "12",
      "wt": "2",
      "atk": "3",
      "def": "2",
      "hp": "5",
      "trait": "Dunedain. Noble. Ranger.",
      "text": "Sentinel.<br />Response: After Aragorn commits to a quest, spend 1 resource from his resource pool to ready him.",
      "shadow": "",
      "flavor": null,
      "set": "Core",
      "setid": "116",
      "setname": "Core Set",
      "num": "001",
      "illustrator": "John Stanko",
      "img": "aragorn-core.jpg",
      "imgfolder": "lotr",
      "imgsize": "high",
      "sequence": "",
      "fullurl": "http://www.cardgamedb.com/index.php/lotr/lord-of-the-rings-card-spoiler/_/core/aragorn-core",
      "guid": "",
      "packquantity": "1",
      "furl": "aragorn-core",
      "rating": "4",
      "max": 1,
      "difficulty": "0",
      "victory": "",
      "banned": null,
      "encounterlist": "",
      "rabbr": ""
    };
    const aragornCard = new PlayerCards['116-001']({name: 'chevalric'}, cardData);
    expect(aragornCard).toMatchSnapshot();
  })
});
