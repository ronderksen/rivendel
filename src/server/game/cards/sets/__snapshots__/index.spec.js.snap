// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlayerCards should be possible to instantiate card with given card data 1`] = `
Aragorn {
  "abilities": Object {
    "actions": Array [],
    "persistentEffects": Array [],
    "playActions": Array [],
    "responses": Array [
      CardResponse {
        "card": [Circular],
        "game": undefined,
        "properties": Object {
          "cost": Object {
            "canPay": [Function],
            "pay": [Function],
          },
          "handler": [Function],
          "when": Object {
            "undefined": [Function],
          },
        },
      },
    ],
  },
  "attack": 3,
  "cardData": Object {
    "atk": "3",
    "banned": null,
    "cost": 0,
    "deck": "Hero",
    "def": "2",
    "difficulty": "0",
    "encounter": "None",
    "encounterlist": "",
    "flavor": null,
    "fullurl": "http://www.cardgamedb.com/index.php/lotr/lord-of-the-rings-card-spoiler/_/core/aragorn-core",
    "furl": "aragorn-core",
    "guid": "",
    "hp": "5",
    "id": "33",
    "illustrator": "John Stanko",
    "img": "aragorn-core.jpg",
    "imgfolder": "lotr",
    "imgsize": "high",
    "label": "Aragorn",
    "max": 1,
    "name": "Aragorn",
    "num": "001",
    "packquantity": "1",
    "rabbr": "",
    "rating": "4",
    "sequence": "",
    "set": "Core",
    "setid": "116",
    "setname": "Core Set",
    "shadow": "",
    "sphere": "Leadership",
    "text": "Sentinel.<br />Response: After Aragorn commits to a quest, spend 1 resource from his resource pool to ready him.",
    "textcost": "",
    "th": "12",
    "trait": "Dunedain. Noble. Ranger.",
    "type": "Hero",
    "unique": "Yes",
    "victory": "",
    "wt": "2",
  },
  "code": "116-001",
  "controller": Object {
    "name": "chevalric",
  },
  "cost": 0,
  "defense": 2,
  "game": undefined,
  "hitPoints": 5,
  "isAttacking": false,
  "isCommittedToQuest": false,
  "isDefending": false,
  "isUnique": true,
  "keywords": Object {
    "simple": Object {
      "sentinel": true,
    },
    "valued": Object {},
  },
  "name": "Aragorn",
  "owner": Object {
    "name": "chevalric",
  },
  "spheres": Array [
    "leadership@spheres",
  ],
  "startingThreat": 12,
  "text": "Sentinel.<br />Response: After Aragorn commits to a quest, spend 1 resource from his resource pool to ready him.",
  "tokens": Object {},
  "traits": Object {
    " Noble": true,
    " Ranger": true,
    "Dunedain": true,
  },
  "type": "hero@cardTypes",
  "uuid": "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx",
  "willPower": 2,
  "wounds": 0,
}
`;
